TACC/tup-ui
# feat(tup-cms): homepage banner

## Overview

- Style homepage banner.
- Add CSS to `tup-cms/static` dir.

## Related

<details><summary>requires Core-CMS support for picture plugin children as caption content</summary>

- https://github.com/TACC/Core-CMS/pull/609
- https://github.com/TACC/Core-CMS/pull/610
- https://github.com/TACC/Core-CMS/commit/71dfb1b

</details>

## Changes

- **changed** core-cms image
- **added** `css-banner.html` snippet* (and sample markup)
- **added** a few small stylesheets (to be integrated into core)†

<sup>* This might be integrated into Core-CMS. Usage requires needs designer review. (Should/Can it instead be one `.banner` class for the container?)</sup>
<sup>† The #174 PR should use this transient method instead.</sup>

## Testing

0. Create a TACC homepage banner.
    <sup>Already built on https://dev.tup.tacc.utexas.edu/. (Figure Layout is visible. Picture + Link Layout is hidden.)</sup>

    <details><summary>Figure Layout</summary>

    > ✅ Easier to build.
    > ℹ️ More markup. Semantic markup.
    1. Create Banner cells using this plugin chain:
        - Column
            - Picture (with a Link)
                - Text (with `<span>` and `<h2>`)
    2. Verify each banner cell is made of this markup:
        `<div>` → `<a>` → `<figure>` → `<img>` & `<figcaption>` → `<span>` & `<h2>`

    </details>
    <details><summary>Picture + Link Layout</summary>

    > ❌ Tedious to build.
    > ❌ Unable to edit Picture via double-click.
    1. Create Banner cells using this plugin chain:
        - Column
            - Picture (no Link)
            - Link
                - Text (with `<span>` and `<h2>`)
    2. Verify each banner cell is made of this markup:
        `<div>` → `<img>` & `<a>` → `<span>` & `<h2>`

    </details>

1. Banner styles are similar* to [design](https://xd.adobe.com/view/dabaec5a-cc31-4c5c-876e-56d6d8cc359e-8fe9/screen/25d576df-4550-4bed-b2a3-5756cb9f2a17/specs/).
2. Each banner "cell" is clickable as one element.
3. The white line between and around banner "cells" is consistently 3px (never 6px).
4. At screen size where large banner heading wraps, verify padding is consistent on all edges of heading text.
5. At screen sizes wider than 576px, verify layout is **flexible** and consistent and nothing pops out of place.
6. At screen sizes narrower than 576px, verify layout is **vertical** and consistent and nothing pops out of place.

<sup>* The original implementation on the site was tweaked live with the designer, and deviates from the design. The stories differ, also. This version is based off of that deviation. The tag colors are missing, because categories (shown as tags) are not confirmed (thus colors not confirmed).</sup>

## UI

### Localhost

| type | render | structure |
| - | - | - |
| figure | ![figure-in-link - render](https://user-images.githubusercontent.com/62723358/225432338-adf8f1da-42ed-41c8-9c64-79ed320da509.png) | ![figure-in-link - structure](https://user-images.githubusercontent.com/62723358/225432332-a218a2a7-7730-4381-8e1c-985455463cce.png) |
| picture + link | ![picture+link - render](https://user-images.githubusercontent.com/62723358/225432339-d730f946-6180-4e41-8969-daa17d02edfd.png) | ![picture+link - structure](https://user-images.githubusercontent.com/62723358/225432337-72995661-dd76-4630-ae1b-521966b24c77.png) |
